local.go

```diff
@@ -138,6 +171,7 @@ func minimize(settings *Settings, location *Location, method Method, funcInfo *F
 
 		// Compute the new function and update the statistics
 		evaluate(funcs, funcInfo, evalType, xNext, location, stats)
+		// TODO: Check for NaN.
```

Comment:
I don't think NaN/Inf is necessarily a problem. Getting it on major iterations is bad, but the linesearchers should be smart enough to backtrack form that (and if they aren't we should make them as such). NaN and Inf are a problem at the starting location because gradients don't make any sense.